lib/ansible/modules/network/nxos/nxos_nxapi.py

```diff
@@ -161,10 +197,24 @@ def check_args(module, warnings):
     return warnings
 
 
-def map_obj_to_commands(want, have, module):
+def map_obj_to_commands(want, have, module, capabilities):
     send_commands = list()
     commands = dict()
 
+    device_info = capabilities.get('device_info')
+    if device_info is None:
+        raise TypeError
```

Comment:
@tstoner I am curious why you are raising `TypeError` if capabilities cannot be fetched. Given that the new options added in this PR are not the required options for `nxos_nxapi` module, I don't think we want to fail the module if device_info is None.
Same for os_version and os_platform. The module can simply ignore the task and pass when these parameters are used if capabilities isn't available instead of failing/raising exception.